### PR TITLE
match glue numbders

### PR DESCRIPTION
**Pull request by liqunfu**
**Description**: Describe your changes.
matching glue between new and old frontend api
**Motivation and Context**
new frontend api

### CODE REVIEW

File: orttraining/orttraining/python/experimental/orttrainer.py [hunk 1]
```diff
@@ -395,41 +397,39 @@ def _combine_torch_model_with_loss_fn(self):
...
+                # model takes input in a order (which can be obtained via inspect.signature(model.forward)) different than input_names.
+                sig = signature(self.model.forward)
+                ordered_list_keys = list(sig.parameters.keys())
 
+                input_dict = {}
+                for key in sig.parameters.keys():
+                    if key in self.input_names_:
+                        input_dict[key] = inputs[self.input_names_.index(key)]
+
+                model_out = self.model(**input_dict)
```
Comment (rayankrish) [inline orttraining/orttraining/python/experimental/orttrainer.py:420]: Will this work as expected if the input is a list? It seems like input_dict would be empty.

Comment (liqunfu) [reply orttraining/orttraining/python/experimental/orttrainer.py:420]: torch.onnx.export always takes dict input.

File: orttraining/orttraining/test/python/orttraining_transformer_trainer.py [hunk 2]
```diff
@@ -20,10 +20,18 @@
 from transformers.modeling_utils import PreTrainedModel
 from transformers.training_args import TrainingArguments
 
+# import sys
```
Comment (spandantiwari) [inline orttraining/orttraining/test/python/orttraining_transformer_trainer.py:23]: Can be removed?

File: orttraining/orttraining/test/python/orttraining_transformer_trainer.py [hunk 3]
```diff
@@ -20,10 +20,18 @@
...
+# sys.path.append("/bert_ort/liqun/onnxruntime/build/Linux/Debug")
+
 import onnxruntime
 from orttraining_test_bert_postprocess import postprocess_model
 from onnxruntime.capi.ort_trainer import ORTTrainer, LossScaler, ModelDescription, IODescription
 
+from onnxruntime.experimental import _utils, amp, optim, orttrainer, TrainStepInfo,\
```
Comment (spandantiwari) [inline orttraining/orttraining/test/python/orttraining_transformer_trainer.py:30]: This can be broken into multiple import statements for formatting and readability.

File: orttraining/orttraining/test/python/orttraining_transformer_trainer.py [hunk 4]
```diff
@@ -81,6 +89,32 @@ def lambda_lr_get_lr(current_global_step):
 
     return lambda_lr_get_lr
 
+def get_const_lr(base_lr):
+    def get_lr(current_step):
+        return base_lr
+    return get_lr
+
+class MyLinearWarmupLRScheduler(_LRScheduler):
```
Comment (spandantiwari) [inline orttraining/orttraining/test/python/orttraining_transformer_trainer.py:97]: Better name for this?

File: orttraining/orttraining/test/python/orttraining_transformer_trainer.py [hunk 5]
```diff
@@ -81,6 +89,32 @@ def lambda_lr_get_lr(current_global_step):
...
+
+    def lr_lambda_linear(self, current_step):
+        if current_step < self.num_warmup_steps:
+            return float(current_step) / float(max(1, self.num_warmup_steps))
+        return max(
+            0.0, float(self.num_training_steps - current_step) / float(max(1, self.num_training_steps - self.num_warmup_steps))
+        )
+
+    def get_lr(self, train_step_info):
+        # LambdaLR increment self.last_epoch at evert sept()
```
Comment (spandantiwari) [inline orttraining/orttraining/test/python/orttraining_transformer_trainer.py:114]: nit:spelling.

File: orttraining/orttraining/test/python/orttraining_transformer_trainer.py [hunk 6]
```diff
@@ -162,31 +199,72 @@ def train(self):
...
-            gradient_accumulation_steps=self.args.gradient_accumulation_steps,
-            world_rank=0, world_size=1,     # only support single GPU cases
-            use_mixed_precision=self.args.fp16,
-            allreduce_post_accumulation=True,
-            get_lr_this_step=get_lr_this_step,
-            loss_scaler=loss_scaler,
-            enable_grad_norm_clip=False,
-            _opset_version=12,
-            _use_deterministic_compute=True)
+        # get_lr_this_step = get_const_lr(self.args.learning_rate)
```
Comment (spandantiwari) [inline orttraining/orttraining/test/python/orttraining_transformer_trainer.py:202]: Remove.

File: orttraining/orttraining/test/python/orttraining_transformer_trainer.py [hunk 7]
```diff
@@ -162,31 +199,72 @@ def train(self):
...
+            params = [{
+                'params': [n for n, p in param_optimizer if "bias" in n or "LayerNorm.weight" in n],
+                "weight_decay_mode": 1, }, {
+                'params': [n for n, p in param_optimizer if not ("bias" in n or "LayerNorm.weight" in n)],
+                "weight_decay_mode": 1, }
+                ]
+
+            optim_config = optim.AdamConfig(params=params,
+                                            lr=2e-5, do_bias_correction=True)
+            # optim_config = optim.AdamConfig(lr=2e-5, do_bias_correction=True)
```
Comment (spandantiwari) [inline orttraining/orttraining/test/python/orttraining_transformer_trainer.py:241]: Remove comment

File: orttraining/orttraining/test/python/orttraining_transformer_trainer.py [hunk 8]
```diff
@@ -258,13 +338,14 @@ def map_optimizer_attributes(name):
         logger.info("\n\nTraining completed. \n\n")
         return TrainOutput(global_step, tr_loss / global_step)
 
-    def _training_step(
-        self, model: ORTTrainer, inputs: Dict[str, torch.Tensor]
-    ) -> float:
+    # def _training_step(
+    #     self, model: ORTTrainer, inputs: Dict[str, torch.Tensor]
+    # ) -> float:
+    def _training_step(self, model, inputs):
```
Comment (spandantiwari) [inline orttraining/orttraining/test/python/orttraining_transformer_trainer.py:344]: Is there a reason the type hints are being removed from the `_training_step` signature?

File: orttraining/orttraining/test/python/orttraining_transformer_trainer.py [hunk 9]
```diff
@@ -313,7 +394,9 @@ def _prediction_loop(
         eval_losses: List[float] = []
         preds: np.ndarray = None
         label_ids: np.ndarray = None
-        self.model.eval()
+
+        if not self.use_new_api:
```
Comment (spandantiwari) [inline orttraining/orttraining/test/python/orttraining_transformer_trainer.py:398]: Is this check enough or does it need a check against torch.no_grad() also before setting the model to eval mode? Typically for f/e use cases we want the model to be in training mode.

Comment (thiagocrepaldi) [reply orttraining/orttraining/test/python/orttraining_transformer_trainer.py:379]: The new frontend does not implement `eval` on ORTTrainer, that is why this is being skipped for the new API

File: orttraining/orttraining/test/python/orttraining_transformer_trainer.py (hunk 8 above)
Comment (thiagocrepaldi) [reply orttraining/orttraining/test/python/orttraining_transformer_trainer.py]: Yes, `model` can be either `onnxruntime.capi.training.ORTTrainer` or `onxnruntime.experimental.ORTTrainer`, which are different types.

File: orttraining/orttraining/test/python/orttraining_transformer_trainer.py [hunk 10]
```diff
@@ -81,6 +89,32 @@ def lambda_lr_get_lr(current_global_step):
 
     return lambda_lr_get_lr
 
+def get_const_lr(base_lr):
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/test/python/orttraining_transformer_trainer.py]: dead code
